Sources/BraintreeThreeDSecure/BTThreeDSecureResult.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+
+#if canImport(BraintreeCore)
+import BraintreeCore
+#endif
+
+#if canImport(BraintreeCard)
+import BraintreeCard
+#endif
+
+#if canImport(BraintreePaymentFlow)
+import BraintreePaymentFlow
+#endif
+
+/// The result of a 3D Secure payment flow
+@objcMembers public class BTThreeDSecureResult: BTPaymentFlowResult {
+
+    // MARK: - Public Properties
+
+    /// The `BTCardNonce` resulting from the 3D Secure flow
+    public var tokenizedCard: BTCardNonce?
+
+    /// The result of a 3D Secure lookup. Contains liability shift and challenge information.
+    public var lookup: BTThreeDSecureLookup?
+
+    /// The error message when the 3D Secure flow is unsuccessful
+    public var errorMessage: String?
+
+    // TODO: this can be internal when BTThreeDSecureAuthenticateJWT is in Swift (was not public so will not need changelog)
+    @objc(initWithJSON:)
+    public init(json: BTJSON? = nil) {
```

Comment:
Same comment - `nil` BTJSON should be handled at the callsite, not in this init.

Reply:
Good call - done: https://github.com/braintree/braintree_ios/pull/988/commits/99ede9ef99f620d78b20bb21345bee7c20ac7302